File: programmers/level1/체육복.java
package programmers;
import java.util.*;
import java.io.*;

public class 체육복 {
	static
	class Solution {
	    public int solution(int n, int[] lost, int[] reserve) {
	        int answer = 0;
	        int[] count = new int[n+2];
	        Arrays.fill(count, 1);
	        count[n+1] = 0;
	        int reserveLength = reserve.length;
	        for(int i = 0; i < reserveLength; ++i) {
	        	count[reserve[i]]++;
	        }
	        int lostLength = lost.length;
	        for(int i = 0; i < lostLength; ++i) {
	        	count[lost[i]]--;
	        }
	        for(int i = 1; i <= n; ++i) {
	        	if(count[i]==0) {
	        		if(count[i-1]>1) {
	        			count[i-1]--;
	        			count[i]++;
	        		}else if(count[i+1]>1) {
	        			count[i+1]--;
	        			count[i]++;
	        		}
	        	}
	        }
	        for(int i = 1; i <= n; ++i) {
	        	if(count[i]>=1) {
	        		answer++;
	        	}
	        }
	        return answer;
	    }
	}
	
	public static void main(String[] args) {
		int n = 5;
		int[] lost = {2,4};
		int[] reserve = {1,3,5};
		Solution sol = new Solution();
		System.out.println(sol.solution(n, lost, reserve));
	}
	
}

File: programmers/js/level1/체육복.js
function solution(n, lost, reserve) {
    var answer = 0;
    let student = [];
    for (let i = 0; i <= n; ++i) {
        student.push(1);
    }
    lost.forEach(e => {
        student[e]--;
    });
    reserve.forEach(e => {
        student[e]++;
    })
    if (student[1] == 0) {
        if (student[2] > 1) {
            student[2]--;
            answer++;
        }
    } else if (student[1] > 0) {
        answer++;
    }
    for (let i = 2; i < n; ++i) {
        if (student[i] == 0) {
            if (student[i - 1] > 1) {
                student[i - 1]--;
                student[i]++;
                answer++;
            } else if (student[i + 1] > 1) {
                student[i + 1]--;
                student[i]++;
                answer++;
            }
        } else if (student[i] > 0) {
            answer++;
        }
    }
    if (student[n] == 0) {
        if (student[n - 1] > 1) {
            answer++;
        }
    } else if (student[n] > 0) {
        answer++;
    }
    return answer;
}

console.log(solution(5, [2, 4], [1, 3, 5]));